src/backend/distributed/commands/alter_table.c

```diff
@@ -1572,3 +1580,95 @@ ExecuteQueryViaSPI(char *query, int SPIOK)
 		ereport(ERROR, (errmsg("could not finish SPI connection")));
 	}
 }
+
+
+/*
+ * SwitchToSequentialAndLocalExecutionIfRelationNameTooLong generates the longest shard name
+ * on the shards of a distributed table, and if exceeds the limit switches to sequential and
+ * local execution to prevent self-deadlocks.
+ *
+ * In case of a RENAME, the relation name parameter should store the new table name, so
+ * that the function can generate shard names of the renamed relations
+ */
+void
```

Comment:
An API that gets both the `relationId` and `relationName` feels confusing -- but not easy to avoid, I can see that/

So, maybe at least can we call it `finalRelationName ` / `newRelationName` or such to indicate that this may not be the name of the input relationId?

Reply:
I changed all occurences with `finalRelationName`. This is a n effective way to prevent potential confusions. Thanks for pointing it out